### PR TITLE
Adding text layers

### PR DESCRIPTION
**Pull request by andymatuschak**
Here’s an API. What do you folks think?

- [ ] add ghpages docs

gif of text layer scene

/cc @jbrennan @may-li @nefaurk @nsfmc @saniul

### CODE REVIEW

**Comment by saniul** [comment]
Yep, this is pretty much how I imagined this.

**Comment by saniul** [comment]
Also, this calls for a text-alignment property.

**Comment by saniul** [comment]
Suggestion, what if we always set `attributedText` under the hood, instead of just `text`? We can also expose an HTML-text setter and use `[[NSAttributedString alloc] initWithData...` internally. Designers can them provide styled strings using that method.

**Comment by saniul** [comment]
https://github.com/alexfish/stylize is also relevant

**Comment by saniul** [comment]
> Also, this calls for a text-alignment property.

Implemented in https://github.com/Khan/Prototope/pull/19

**Comment by andymatuschak** [comment]
Wooo, thanks for making the text alignment pull request! Will take a look at that shortly.

I'd like to wait on exposing fancier styling until we have a better idea of how we'd use it. Would prefer something more structured than HTML (Stylize is a nice reference), but maybe HTML's the easiest/familiar thing. @may-li, @nefaurk, what do you think?

File: PrototopeJSBridge/VideoLayerBridge.swift [hunk 1]
```diff
@@ -11,26 +11,25 @@ import Prototope
...
-	required public init(args: NSDictionary) {
+	required public init?(args: NSDictionary) {
 		let parentLayer = (args["parent"] as! LayerBridge?)?.layer
 		let video = (args["video"] as! VideoBridge?)?.video
 		
-		self.videoLayer = VideoLayer(parent: parentLayer, video: video)
-
-		super.init(args: args)
+		let videoLayer = VideoLayer(parent: parentLayer, video: video)
+		super.init(videoLayer)
```
Comment (andymatuschak) [inline PrototopeJSBridge/VideoLayerBridge.swift]: @jbrennan Here and in ScrollLayerBridge, I made this fix: the previous code actually ended up initializing two layers—the video layer and also the layer in super's `init` implementation.

Implementation inheritance. *sigh*

**Comment by andymatuschak** [comment]
Oh, uh, I totally forgot the ghpages docs. Will do that shortly.

File: PrototopeJSBridge/VideoLayerBridge.swift (hunk 1 above)
Comment (jbrennan) [inline PrototopeJSBridge/VideoLayerBridge.swift]: Oh deuce.

**Comment by nsfmc** [comment]
my only comment, mostly for docs, is that there should be an easy way to know where the 1st baseline goes for a given textlayer (i.e. is it the layer's y minus the default leading plus the descender?)

**Comment by saniul** [comment]
@nsfmc I'll do this tonight when I get home

**Comment by saniul** [comment]
^ https://github.com/Khan/Prototope/pull/24

**Pull request merged**